src/kolibri_android/kolibri_utils.py

```diff
@@ -0,0 +1,139 @@
+    _init_kolibri_env(content_fallback_dirs)
+
+    from kolibri.utils.main import initialize
+
+    for plugin_name in DISABLED_PLUGINS:
+        _disable_kolibri_plugin(plugin_name)
+
+    for plugin_name in REQUIRED_PLUGINS:
+        _enable_kolibri_plugin(plugin_name)
+
+    for plugin_name in OPTIONAL_PLUGINS:
+        _enable_kolibri_plugin(plugin_name, optional=True)
+
+    initialize(**kwargs)
+
+
+def _monkeypatch_whitenoise():
+    from .android_whitenoise import monkeypatch_whitenoise
+
+    monkeypatch_whitenoise()
+
+
+def _init_kolibri_env(content_fallback_dirs=None):
+    from .android_utils import get_endless_key_uris
+    from .android_utils import get_android_node_id
+    from .android_utils import get_home_folder
+    from .android_utils import get_signature_key_issuing_organization
+    from .android_utils import get_timezone_name
+    from .android_utils import get_version_name
+    from .android_whitenoise import DynamicWhiteNoise
+
+    signing_org = get_signature_key_issuing_organization()
+    if signing_org == "Learning Equality":
+        runmode = "android-testing"
+    elif signing_org == "Android":
+        runmode = "android-debug"
+    elif signing_org == "Google Inc.":
+        runmode = ""  # Play Store!
+    else:
+        runmode = "android-" + re.sub(r"[^a-z ]", "", signing_org.lower()).replace(
+            " ", "-"
+        )
+    os.environ["KOLIBRI_RUN_MODE"] = runmode
+    os.environ["KOLIBRI_PROJECT"] = "endless-key-android"
+
+    os.environ["TZ"] = get_timezone_name()
+    os.environ["LC_ALL"] = "en_US.UTF-8"
+
+    os.environ["KOLIBRI_HOME"] = get_home_folder()
+
+    endless_key_uris = get_endless_key_uris()
+
+    if endless_key_uris is not None:
+        content_fallback_dirs = DynamicWhiteNoise.encode_root(
+            endless_key_uris["content"]
+        )
+        logging.info(
+            "Setting KOLIBRI_CONTENT_FALLBACK_DIRS to %s", content_fallback_dirs
+        )
+        os.environ["KOLIBRI_CONTENT_FALLBACK_DIRS"] = content_fallback_dirs
+
+    os.environ["KOLIBRI_APK_VERSION_NAME"] = get_version_name()
+    os.environ["DJANGO_SETTINGS_MODULE"] = "kolibri_android.kolibri_extra.settings"
+
+    AUTOPROVISION_PATH = SCRIPT_PATH.joinpath("automatic_provision.json")
+    if AUTOPROVISION_PATH.is_file():
+        os.environ["KOLIBRI_AUTOMATIC_PROVISION_FILE"] = AUTOPROVISION_PATH.as_posix()
+
+    os.environ["KOLIBRI_CHERRYPY_THREAD_POOL"] = "2"
+
+    os.environ["KOLIBRI_APPS_BUNDLE_PATH"] = SCRIPT_PATH.joinpath(
+        "apps-bundle", "apps"
+    ).as_posix()
+    os.environ["KOLIBRI_CONTENT_COLLECTIONS_PATH"] = SCRIPT_PATH.joinpath(
+        "collections"
+    ).as_posix()
+
+    node_id = get_android_node_id()
+
+    # Don't set this if the retrieved id is falsy, too short, or a specific
+    # id that is known to be hardcoded in many devices.
+    if node_id and len(node_id) >= 16 and node_id != "9774d56d682e549c":
+        os.environ["MORANGO_NODE_ID"] = node_id
```

Comment:
Can you explain what is this about? Ideally it should be in a separate commit as this is new stuff not a refactor. What is affected by setting the `MORANGO_NODE_ID` environment variable? I don't see it in Kolibri's option.ini or anywhere in its source code.

Reply:
This comes from [initialization.py](https://github.com/endlessm/kolibri-installer-android/blob/develop/src/initialization.py#L77-L80). This particular thing was added in learningequality/kolibri-installer-android#88, and it causes Kolibri's database replication system to reuse the Android device ID instead of its default behaviour. I will add another commit that is just moving stuff out of initialize.py so that the provenance of stuff like this is clearer in the commit history.